flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/source/TestingSourceOperator.java

```diff
@@ -86,6 +87,7 @@ public TestingSourceOperator(
 		this.parallelism = parallelism;
 		this.metrics = UnregisteredMetricGroups.createUnregisteredOperatorMetricGroup();
 		this.readerCreated = false;
+		this.sourceMetricGroup = new TestingSourceMetricGroup();
```

Comment:
What do you think about having a testing constructor in the `SourceOperator` that accepts a metric group, rather making a parent field public and mutating it from here?